atomicapp/providers/lib/kubeconfig.py

```diff
@@ -103,16 +106,52 @@ def parse_kubeconf_data(kubecfg):
         logger.debug("user: %s", user)
 
         url = cluster["cluster"]["server"]
-        token = user["user"]["token"]
-        if "namespace" in context["context"]:
-            namespace = context["context"]["namespace"]
-        if "insecure-skip-tls-verify" in cluster["cluster"]:
-            tls_verify = not cluster["cluster"]["insecure-skip-tls-verify"]
-        elif "certificate-authority" in cluster["cluster"]:
-            ca = cluster["cluster"]["certificate-authority"]
+        auth = user["user"].get("token")
+        namespace = context["context"].get("namespace")
+        tls_verify = not cluster["cluster"].get("insecure-skip-tls-verify")
+
+        if tls_verify:
+            ca_data = cluster["cluster"].get("certificate-authority-data")
+            if ca_data:
+                ca = Utils.getTmpFile(b64decode(ca_data))
+            else:
+                if "certificate-authority" in cluster["cluster"]:
+                    # if we are in container translate path to path on host
+                    ca = os.path.join(Utils.getRoot(),
+                                      cluster["cluster"].get("certificate-authority").lstrip('/'))
+
+        if not auth:
+            # If token not specified, check for certificate auth.
+
+            # client-certificate-data and client-key-data options overrides
+            # client-certificate and client-key
+            # https://github.com/kubernetes/kubernetes/blob/v1.2.2/pkg/client/unversioned/clientcmd/api/types.go#L78
+            # `{client-certificate,client-key}-data` keys in Kubernetes config
+            # file are inline base64 encoded certificates, requests library
+            # requires certs in files, this is why we are putting them to tmp
+            # files.
+
+            cert_data = user["user"].get("client-certificate-data")
+            key_data = user["user"].get("client-key-data")
+
+            if cert_data:
+                cert = Utils.getTmpFile(b64decode(cert_data))
```

Comment:
Do these tmp files ever get cleaned up? If not we should probably try to make sure they do. I know this makes it harder.